www/js/metrics/metricsHelper.ts

```diff
@@ -0,0 +1,212 @@
+  }
+  return greaterLesserPct;
+}
+
+export function parseDataFromMetrics(metrics, population) {
+  console.log("Called parseDataFromMetrics on ", metrics);
+  let mode_bins = {};
+  metrics?.forEach(function(metric) {
+    let onFootVal = 0;
+
+    for (let field in metric) {
+      /*For modes inferred from sensor data, we check if the string is all upper case 
+      by converting it to upper case and seeing if it is changed*/
+      if(field == field.toUpperCase()) {
+        /*sum all possible on foot modes: see https://github.com/e-mission/e-mission-docs/issues/422 */
+        if (isOnFoot(field)) {
+          onFootVal += metricToValue(population, metric, field);
+          field = 'ON_FOOT';
+        }
+        if (!(field in mode_bins)) {
+          mode_bins[field] = [];
+        }
+        //for all except onFoot, add to bin - could discover mult onFoot modes
+        if (field != "ON_FOOT") {
+          mode_bins[field].push([metric.ts, metricToValue(population, metric, field), metric.fmt_time]);
+        }
+      }
+      //this section handles user lables, assuming 'label_' prefix
+      if(field.startsWith('label_')) {
+        let actualMode = field.slice(6, field.length); //remove prefix
+        console.log("Mapped field "+field+" to mode "+actualMode);
+        if (!(actualMode in mode_bins)) {
+            mode_bins[actualMode] = [];
+        }
+        mode_bins[actualMode].push([metric.ts, Math.round(metricToValue(population, metric, field)), moment(metric.fmt_time).format()]);
+      }
+    }
+    //handle the ON_FOOT modes once all have been summed
+    if ("ON_FOOT" in mode_bins) {
+      mode_bins["ON_FOOT"].push([metric.ts, Math.round(onFootVal), metric.fmt_time]);
+    }
+  });
+
+  let return_val = [];
+  for (let mode in mode_bins) {
+      return_val.push({key: mode, values: mode_bins[mode]});
+  }
+
+  return return_val;
+}
+
+export function generateSummaryFromData(modeMap, metric) {
+  console.log("Invoked getSummaryDataRaw on ", modeMap, "with", metric);
+
+  let summaryMap = [];
+
+  for (let i=0; i < modeMap.length; i++){
+    let summary = {};
+    summary['key'] = modeMap[i].key; 
+    let sumVals = 0;
+
+    for (let j = 0; j < modeMap[i].values.length; j++)
+    {
+      sumVals += modeMap[i].values[j][1]; //2nd item of array is value
+    }
+    if (metric === 'mean_speed'){ 
+      //we care about avg speed, sum for other metrics
+      summary['values'] = Math.round(sumVals / modeMap[i].values.length);
+    } else {
+      summary['values'] = Math.round(sumVals);
+    }
+
+    summaryMap.push(summary);
+  }
+
+  return summaryMap;
+}
+
+/*
+* We use the results to determine whether these results are from custom
+* labels or from the automatically sensed labels. Automatically sensedV
+* labels are in all caps, custom labels are prefixed by label, but have had
+* the label_prefix stripped out before this. Results should have either all
+* sensed labels or all custom labels.
+*/
```

Comment:
for the record: this should eventually go into the dynamic config - whether people want to use sensed values or user values. Or it may be moot once "count every trip" is implemented.